### PR TITLE
print argument enums in help message

### PR DESCRIPTION
**Pull request by mirpedrol**
Always print the options of an argument with the help message. 

The output with too many options which are chopped looks like this:
Screenshot 2022-12-13 at 13 21 16

### CODE REVIEW

File: plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy [hunk 1]
```diff
@@ -257,9 +257,27 @@ class SchemaValidator extends PluginExtensionPoint {
...
+                        if (add_val.length() > dec_linewidth - 5) {
+                            chop_enums += ", ..."
+                        } else {
+                            if (chop_enums == "") {
+                                chop_enums += val
+                            } else {
+                                chop_enums += ", " + val
+                            }
+                        }
+                    }
```
Comment (ewels) [inline plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy:275]: Feels a bit verbose - I wonder if we can do basically the same with a join? Something like this: (untested)
```suggestion
                    def String chop_enums = enums.join(", ")
                    if(chop_enums.length() > dec_linewidth){
                        chop_enums = chop_enums.substring(0, dec_linewidth-5) + ", ..."
                    }
```

File: plugins/nf-validation/src/test/nextflow/validation/PluginExtensionMethodsTest.groovy [hunk 2]
```diff
@@ -401,6 +401,33 @@ class PluginExtensionMethodsTest extends Dsl2Spec{
...
+            log.info help_msg
+        """
+
+        when:
+        dsl_eval(SCRIPT_TEXT)
+        def stdout = capture
+                .toString()
+                .readLines()
+                .findResults {it.contains('publish_dir_mode') && 
+                    it.contains('(accepted: symlink, rellink, link, copy, copyNoFollow')
```
Comment (ewels) [inline plugins/nf-validation/src/test/nextflow/validation/PluginExtensionMethodsTest.groovy:423]:
```suggestion
                    it.contains('(accepted: symlink, rellink, link, copy, copyNoFollow)') 
```

**Review by ewels — APPROVED** [review]
Nice!

File: plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy (hunk 1 above)
Comment (mirpedrol) [reply plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy:275]: I tried to use `.join()` but didn't manage to chop the string as I wanted, I will try your suggestion, thanks!

File: plugins/nf-validation/src/test/nextflow/validation/PluginExtensionMethodsTest.groovy (hunk 2 above)
Comment (ewels) [reply plugins/nf-validation/src/test/nextflow/validation/PluginExtensionMethodsTest.groovy:423]: hah, fair enough..

**Review by ewels — APPROVED** [review]
Merge conflict, but then good to go

**Comment by mirpedrol** [comment]
Merging conflicts are resolved, could you merge it? Thanks!

**Pull request merged**